api/restHandler/BulkUpdateRestHandler.go

```diff
@@ -0,0 +1,193 @@
+	enforcerUtil rbac.EnforcerUtil, envService request.EnvironmentService,
+	gitRegistryConfig pipeline.GitRegistryConfig, dockerRegistryConfig pipeline.DockerRegistryConfig,
+	cdHandelr pipeline.CdHandler,
+	appCloneService appClone.AppCloneService,
+	appWorkflowService appWorkflow.AppWorkflowService,
+	materialRepository pipelineConfig.MaterialRepository, policyService security2.PolicyService,
+	scanResultRepository security.ImageScanResultRepository) *BulkUpdateRestHandlerImpl {
+	return &BulkUpdateRestHandlerImpl{
+		pipelineBuilder:         pipelineBuilder,
+		Logger:                  Logger,
+		chartService:            chartService,
+		propertiesConfigService: propertiesConfigService,
+		dbMigrationService:      dbMigrationService,
+		application:             application,
+		userAuthService:         userAuthService,
+		validator:               validator,
+		teamService:             teamService,
+		enforcer:                enforcer,
+		ciHandler:               ciHandler,
+		gitSensorClient:         gitSensorClient,
+		ciPipelineRepository:    ciPipelineRepository,
+		pipelineRepository:      pipelineRepository,
+		enforcerUtil:            enforcerUtil,
+		envService:              envService,
+		gitRegistryConfig:       gitRegistryConfig,
+		dockerRegistryConfig:    dockerRegistryConfig,
+		cdHandelr:               cdHandelr,
+		appCloneService:         appCloneService,
+		appWorkflowService:      appWorkflowService,
+		materialRepository:      materialRepository,
+		policyService:           policyService,
+		scanResultRepository:    scanResultRepository,
+	}
+}
+
+func (handler BulkUpdateRestHandlerImpl) GetExampleOperationBulkUpdate(w http.ResponseWriter, r *http.Request) {
+	response, err := handler.chartService.GetBulkUpdateRequestExample("deployment-template")
+	if err != nil {
+		writeJsonResp(w, err, nil, http.StatusInternalServerError)
+		return
+	}
+
+	//auth free, only login required
+	var responseArr []pipeline.BulkUpdateRequest
+	responseArr = append(responseArr, response)
+	writeJsonResp(w, nil, responseArr, http.StatusOK)
+}
+func (handler BulkUpdateRestHandlerImpl) GetAppNameDeploymentTemplate(w http.ResponseWriter, r *http.Request) {
+	decoder := json.NewDecoder(r.Body)
+	var script pipeline.BulkUpdateScript
+	err := decoder.Decode(&script)
+	if err != nil {
+		writeJsonResp(w, err, nil, http.StatusBadRequest)
+		return
+	}
+	err = handler.validator.Struct(script)
+	if err != nil {
+		handler.Logger.Errorw("validation err, Script", "err", err, "BulkUpdateScript", script)
+		writeJsonResp(w, err, nil, http.StatusBadRequest)
+		return
+	}
+	token := r.Header.Get("token")
+	impactedApps, err := handler.chartService.GetBulkAppName(script.Payload)
+	if err != nil {
+		writeJsonResp(w, err, nil, http.StatusInternalServerError)
+		return
+	}
+	for _, impactedApp := range impactedApps {
+		resourceName := handler.enforcerUtil.GetAppRBACName(impactedApp.AppName)
```

Comment:
this may have performance implications

Reply:
will extract out, pull in single query. @kartik-579 i'll fix this.